nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -45,65 +45,153 @@ def _setup_tokenizer(self, tokenizer_cfg: DictConfig):
             else:
-                model_path = os.path.join(self.tokenizer_dir, 'tokenizer.model')
-            model_path = self.register_artifact('tokenizer.model_path', model_path)
-            self.model_path = model_path
+                model_path = self.register_artifact('tokenizer.model_path', model_path)
 
-            if 'special_tokens' in self.tokenizer_cfg:
-                special_tokens = self.tokenizer_cfg['special_tokens']
+            if 'special_tokens' in tokenizer_cfg:
+                special_tokens = tokenizer_cfg['special_tokens']
 
                 if special_tokens is not None:
-                    raise ValueError("`special_tokens` are no longer supported for SentencePiece based tokenizers.")
+                    raise ValueError('`special_tokens` are no longer supported for SentencePiece based tokenizers.')
 
             # Update special tokens
-            self.tokenizer = tokenizers.SentencePieceTokenizer(model_path=model_path)
+            tokenizer = tokenizers.SentencePieceTokenizer(model_path=model_path)
+
+            if 'vocab_path' in tokenizer_cfg:
+                vocab_path = tokenizer_cfg.get('vocab_path')
+            else:
+                vocab_path = os.path.join(tokenizer_dir, 'vocab.txt')
 
-            if 'vocab_path' in self.tokenizer_cfg:
-                vocab_path = self.tokenizer_cfg.get('vocab_path')
+            if lang is not None:
+                vocab_path = self.register_artifact('tokenizer.tokenizers.' + lang + '.vocab_path', vocab_path)
             else:
-                vocab_path = os.path.join(self.tokenizer_dir, 'vocab.txt')
-            vocab_path = self.register_artifact('tokenizer.vocab_path', vocab_path)
-            self.vocab_path = vocab_path
+                vocab_path = self.register_artifact('tokenizer.vocab_path', vocab_path)
 
             try:
-                if 'spe_tokenizer_vocab' in self.tokenizer_cfg:
-                    spe_vocab_path = self.tokenizer_cfg.get('spe_tokenizer_vocab')
+                if 'spe_tokenizer_vocab' in tokenizer_cfg:
+                    spe_vocab_path = tokenizer_cfg.get('spe_tokenizer_vocab')
+                else:
+                    spe_vocab_path = os.path.join(tokenizer_dir, 'tokenizer.vocab')
+
+                if lang is not None:
+                    spe_vocab_path = self.register_artifact(
+                        'tokenizer.tokenizers.' + lang + '.spe_tokenizer_vocab', spe_vocab_path,
+                    )
                 else:
-                    spe_vocab_path = os.path.join(self.tokenizer_dir, 'tokenizer.vocab')
-                spe_vocab_path = self.register_artifact('tokenizer.spe_tokenizer_vocab', spe_vocab_path)
-                self.spe_vocab_path = spe_vocab_path
+                    spe_vocab_path = self.register_artifact('tokenizer.spe_tokenizer_vocab', spe_vocab_path)
+
             except FileNotFoundError:
```

Comment:
Missed self.spe_vocab_path assignment

Reply:
can't make the assignment because there is more than one potentially; this method just builds, does not assign